webapp/src/main/java/com/box/l10n/mojito/service/repository/statistics/RepositoryStatisticService.java

```diff
@@ -134,10 +127,10 @@ public void updateStatistics(Long repositoryId) {
       repositoryStatisticRepository.save(repositoryStatistic);
 
       logger.debug("Update locale statistics");
-      for (RepositoryLocale repositoryLocale :
-          repositoryService.getRepositoryLocalesWithoutRootLocale(repository)) {
-        updateLocaleStatistics(repositoryLocale, repositoryStatistic);
-      }
+
+      repositoryService.getRepositoryLocalesWithoutRootLocale(repository).parallelStream()
```

Comment:
since this is IO heavy, I think we should have a dedicated thread pool. I don't think the parallelStream one is configurable though. So it may take a bit more refactoring.

Reply:
@aurambaj - unless this came up as a problem in local testing, I'd be in favour of rolling out this change and only adding a dedicated thread pool if we explicitly see that there is a need for it, as otherwise that optimisation and extra refactoring might not be required at all - it could turn out we could be running a lot more of those operations in parallel in addition to these without any proble,.

The key thing that is needed is reliable monitoring for thread starvation - which will help us make that decision.

Reply:
We chatted with @maallen and it felt it was better to have a dedicated thread pool. AFAIK, we're not using parrallelstream() anywhere else, but the underlying threadpool might be used and we'd rather have an identified impact with that change. But I also see your point, don't fix it if it is not broken, ... not sure we want to figure it out later though

Reply:
Updated the approach to use a task executor with configurable core and max pool sizes.